### PR TITLE
Closes #246

### LINKED ISSUE #246: Create dataset loader for PsyTAR
## Adding a Dataset
- **Name:** PsyTAR
- **Description:** *None provided*
- **Task:** NER|SENT_CLASS
- **Paper:** https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6495095/
- **Data:** https://www.askapatient.com/research/pharmacovigilance/corpus-ades-psychiatric-medications.asp
- **License:** CC BY 4.0

### PR DESCRIPTION
**Pull request by danilexn**
Closes #246 

- **Description:** PsyTAR dataset contains 891 drugs reviews posted by patients on "askapatient.com", about the effectiveness and adverse drug events associated with Zoloft, Lexapro, Cymbalta, and Effexor XR.

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.

### CODE REVIEW

**Comment by danilexn** [comment]
Dear reviewers, please consider:

- The dataset is local
- Tests pass, but there are many wrong offset warnings. It seems that the dataset contains many differences between the entity's texts and the text by offset, because of different casing; for example: `text:'depression' != text_by_offset:'DEPRESSION'`
- The dataset is in XLSX format; therefore, pandas was used for reading these files. However, creating the development environment following the instructions in [CONTRIBUTING.md](https://github.com/bigscience-workshop/biomedical/blob/940af66aee80c4f004bf33d199690a511b08f8c8/CONTRIBUTING.md) does not seem to install `openpyxl`.

Thanks for your time!

**Comment by hakunanatasha** [comment]
Hi @danilexn I was able to download the dataset, but I'm unable to run the unit tests for your script; can you let me know the exact line you ran for them?

**Comment by danilexn** [comment]
> Hi @danilexn I was able to download the dataset, but I'm unable to run the unit tests for your script; can you let me know the exact line you ran for them?

Thanks for your reply, @hakunanatasha 
I ran the tests with

`python -m tests.test_bigbio biodatasets/psytar/psytar.py --data_dir /home/danilexn/Desktop/PsyTAR_dataset.xlsx`

As I pointed out in the comment above, is `openpyxl` available in your environment?

**Comment by hakunanatasha** [comment]
Hi @danilexn sorry for the slow down - this PR is almost ready to merge but I noticed one caveat that I can't seem to figure out internally.

If I specify the config, as such:
```
from datasets import load_dataset
x = load_dataset("biodatasets/psytar/psytar/psytar.py", data_dir="/path/to/file", name="psytar_source") # or the bigbio view
```
It works fine (and passes unit-tests)

But if I do:
```
from datasets import load_dataset
x = load_dataset("biodatasets/psytar/psytar/psytar.py", data_dir="/path/to/file") # or the bigbio view
```
it does not work.

This is strange since the script looks correct, and you've defined a default config correctly as source. I don't know if it's something to do with HF's internal dataset builder when a specific dataset path is provided. 

Would you be willing to look into this? If you can't figure it out, let's merge with this caveat, as it still passes the expectations of a dataloader.

**Comment by galtay** [comment]
@hakunanatasha @danilexn I've had this same issue (also with a local dataset). 

this works,
```
ds = load_dataset("biodatasets/n2c2_2008/n2c2_2008.py", name="n2c2_2008_source", data_dir="/Users/gabriel.altay/Data/bigbio/n2c2-2008")
``` 

while this fails,
```
ds = load_dataset("biodatasets/n2c2_2008/n2c2_2008.py", data_dir="/Users/gabriel.altay/Data/bigbio/n2c2-2008")
```

even though the `DEFAULT_CONFIG_NAME = "n2c2_2008_source"`

I think either this is a bug in `load_dataset` or we are not understanding the `load_dataset` function fully. Not sure exactly what the problem is, but I don't think it's particular to this dataset.

**Comment by danilexn** [comment]
[2e91559](https://github.com/bigscience-workshop/biomedical/pull/364/commits/2e915599dcbe966d2b3b6525cdb53dacadecb8c3) passes all tests and loads the source schema when no `name` is specified in `load_dataset`

**Comment by hakunanatasha** [comment]
Wooohoo! I’ll merge this today

File: biodatasets/psytar/psytar.py [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+_BIGBIO_VERSION = "1.0.0"
+
+
+@dataclass
+class PsyTARBigBioConfig(datasets.BuilderConfig):
+    schema: str = "source"
+    name: str = "psytar_source"
+    version: datasets.Version = _SOURCE_VERSION
+    description: str = "PsyTAR source schema"
+    subset_id: str = "psytar"
```
Comment (galtay) [inline biodatasets/psytar/psytar.py:112]: my understanding is that you still want to `from utils.configs import BigBioConfig` and use it in `BUILDER_CONFIGS` below

File: biodatasets/psytar/psytar.py [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+        PsyTARBigBioConfig(
+            name="psytar_bigbio_text",
+            version=BIGBIO_VERSION,
+            description="PsyTAR BigBio text classification schema",
+            schema="bigbio_text",
+            subset_id="psytar",
+        ),
+    ]
+
+    BUILDER_CONFIG_CLASS = PsyTARBigBioConfig
```
Comment (galtay) [inline biodatasets/psytar/psytar.py:146]:
```suggestion
    BUILDER_CONFIG_CLASS = BigBioConfig
```

Comment (danilexn) [reply biodatasets/psytar/psytar.py:146]: Sure, I'm going to consider these changes once we get some feedback from @lhoestq 

Thanks!

**Comment by galtay** [comment]
> The `DEFAULT_CONFIG_NAME` is only used to pick the default config from the `BUILDER_CONFIGS` list if no parameters are passed. Otherwise it instantiantes a new configuration using the class specified in `BUILDER_CONFIG_CLASS`.
> 
> I think you can subclass `BigBioConfig` to `PsyTARBigBioConfig ` and set a default `name` if you want Then you can keep `BUILDER_CONFIG_CLASS = PsyTARBigBioConfig`.

I think I understand now, thanks for the input @lhoestq ! The solution above is a good one and we might do that for each dataset loader after the hackathon is done, but I think for now I may add a `__post_init__` method to our `BigBioConfig` that makes `name` a required arg so that its always explicit.

**Review by hakunanatasha — APPROVED** [review]

**Pull request merged**